pkg/apis/provisioning/v0alpha1/types.go

```diff
@@ -68,36 +68,67 @@ const (
 )
 
 type RepositorySpec struct {
-	// Describe the feature toggle
+	// The repository display name (shown in the UI)
 	Title string `json:"title"`
 
-	// Describe the feature toggle
+	// Repository description
 	Description string `json:"description,omitempty"`
 
-	// The folder that is backed by the repository.
-	// The value is a reference to the Kubernetes metadata name of the folder in the same namespace.
-	Folder string `json:"folder,omitempty"`
-
 	// Edit options within the repository
 	Editing EditingOptions `json:"editing"`
 
+	// Sync settings -- how values are pulled from the repository into grafana
+	Sync SyncOptions `json:"sync"`
+
 	// The repository type.  When selected oneOf the values below should be non-nil
 	Type RepositoryType `json:"type"`
 
 	// The repository on the local file system.
-	// Mutually exclusive with s3 and github.
+	// Mutually exclusive with local | s3 | github.
 	Local *LocalRepositoryConfig `json:"local,omitempty"`
 
 	// The repository in an S3 bucket.
-	// Mutually exclusive with local and github.
+	// Mutually exclusive with local | s3 | github.
 	S3 *S3RepositoryConfig `json:"s3,omitempty"`
 
 	// The repository on GitHub.
-	// Mutually exclusive with local and s3.
+	// Mutually exclusive with local | s3 | github.
 	// TODO: github or just 'git'??
 	GitHub *GitHubRepositoryConfig `json:"github,omitempty"`
 }
 
+// SyncTargetType defines where we want all values to resolve
+// +enum
+type SyncTargetType string
+
+// RepositoryType values
+const (
+	// Resources should be saved into the root grafana directory
+	// currently, only one repository may specify `root` target
+	SyncTargetTypeRoot SyncTargetType = "root"
+
+	// Resources will be saved into a folder managed by this repository
+	SyncTargetTypeFolder SyncTargetType = "folder"
+
+	// Pick an explicit folder where we can save resources, not managed by provisioning
+	// SyncTargetTypeShared SyncTargetType = "shared"
```

Comment:
we don't need this one yet -- but essentially a version where repositories could dump into a folder that it does not "own"

Reply:
🤔   Alternative naming:

* instance  -- the entire grafana instance
* root = an automatic root folder
* folder (not yet!) an explicit folder that may not be owned by this repository

Reply:
I like the `instance` part but I am not sure if it will be confusing to call `root` to the other one. 

Alternative: 

instance -- the entire grafana instance
root = an automatic root folder
custom (not yet!) an explicit folder that may not be owned by this repository

Reply:
Lets try "instance" | "mirror" and see how it goes :)